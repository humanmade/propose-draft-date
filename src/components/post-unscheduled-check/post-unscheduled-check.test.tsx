import React from 'react';
import {
	cleanup,
	render,
	RenderResult,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import {
	PostUnscheduledCheck,
	PostUnscheduledCheckProps,
} from './index';

describe( 'PostUnscheduledCheck', () => {
	let renderWithProps: (
		props: PostUnscheduledCheckProps,
	) => RenderResult;

	beforeEach( () => {
		renderWithProps = ( props ) => render( (
			<PostUnscheduledCheck { ...props }>
				<p>Wrapped Content</p>
			</PostUnscheduledCheck>
		) );
	} );

	afterEach( cleanup );

	it( 'returns null if a post doe not have a "floating" date', () => {
		const { container, queryByText } = renderWithProps( {
			isFloating: false,
			hasPublishAction: false,
			isPublished: false,
		} );
		expect( container ).toMatchInlineSnapshot( '<div />' );
		expect( queryByText( 'Wrapped Content' ) ).toBeNull();
	} );

	it( 'returns null if the user is capable of publishing posts', () => {
		const { container, queryByText } = renderWithProps( {
			isFloating: true,
			hasPublishAction: true,
			isPublished: false,
		} );
		expect( container ).toMatchInlineSnapshot( '<div />' );
		expect( queryByText( 'Wrapped Content' ) ).toBeNull();
	} );

	it( 'returns null if isPublished is true', () => {
		const { container, queryByText } = renderWithProps( {
			isFloating: true,
			hasPublishAction: false,
			isPublished: true,
		} );
		expect( container ).toMatchInlineSnapshot( '<div />' );
		expect( queryByText( 'Wrapped Content' ) ).toBeNull();
	} );

	it( 'returns children if all conditions are satisfied', () => {
		const { container, queryByText } = renderWithProps( {
			isFloating: true,
			hasPublishAction: false,
			isPublished: false,
		} );
		expect( container ).toMatchSnapshot();
		expect( queryByText( 'Wrapped Content' ) ).toBeInTheDocument();
	} );
} );
