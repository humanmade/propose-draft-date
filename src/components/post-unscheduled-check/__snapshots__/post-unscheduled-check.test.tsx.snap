// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostUnscheduledCheck returns children if all conditions are satisfied when post status is auto-draft 1`] = `
<div>
  <p>
    Wrapped Content
  </p>
</div>
`;

exports[`PostUnscheduledCheck returns children if all conditions are satisfied when post status is draft 1`] = `
<div>
  <p>
    Wrapped Content
  </p>
</div>
`;

exports[`PostUnscheduledCheck returns children if all conditions are satisfied when post status is future 1`] = `
<div>
  <p>
    Wrapped Content
  </p>
</div>
`;
